test/test_console_tool.py

```diff
@@ -121,13 +127,72 @@ def test_buckets(self):
         {
             "accountId": "my-account",
             "bucketId": "bucket_1",
+            "bucketInfo": {},
             "bucketName": "your-bucket",
-            "bucketType": "allPrivate"
+            "bucketType": "allPrivate",
+            "lifecycleRules": [],
+            "revision": 1
         }
         '''
 
         self._run_command(['delete_bucket', 'your-bucket'], expected_stdout, '', 0)
 
+    def test_bucket_info_from_file(self):
+
+        self._authorize_account()
+        self._run_command(['create_bucket', 'my-bucket', 'allPublic'], 'bucket_0\n', '', 0)
+
+        with TempDir() as temp_dir:
+            bucket_info = {'color': 'blue'}
+            bucket_info_file = self._make_local_file(temp_dir, 'bucket_info.json')
+            with open(bucket_info_file, 'wb') as f:
+                f.write(six.b(json.dumps(bucket_info)))
+
+            expected_stdout = '''
+                {
+                    "accountId": "my-account",
+                    "bucketId": "bucket_0",
+                    "bucketInfo": {
+                        "color": "blue"
+                    },
+                    "bucketName": "my-bucket",
+                    "bucketType": "allPrivate",
+                    "lifecycleRules": [],
+                    "revision": 2
+                }
+                '''
+            self._run_command(
+                [
+                    'update_bucket', '--bucketInfo', '@' + bucket_info_file, 'my-bucket',
+                    'allPrivate'
+                ], expected_stdout, '', 0
+            )
+
+    def test_bucket_info_from_json(self):
+
+        self._authorize_account()
+        self._run_command(['create_bucket', 'my-bucket', 'allPublic'], 'bucket_0\n', '', 0)
+
+        bucket_info = {'color': 'blue'}
+
+        expected_stdout = '''
+            {
+                "accountId": "my-account",
+                "bucketId": "bucket_0",
+                "bucketInfo": {
+                    "color": "blue"
+                },
+                "bucketName": "my-bucket",
+                "bucketType": "allPrivate",
+                "lifecycleRules": [],
+                "revision": 2
+            }
+            '''
+        self._run_command(
+            ['update_bucket', '--bucketInfo', json.dumps(bucket_info), 'my-bucket', 'allPrivate'],
+            expected_stdout, '', 0
+        )
+
```

Comment:
The `b2_fuse` project will use it over the (so far supported) API level as soon as it is released, which may be before command line support is added. Therefore I think this should be tested with corner-case scenarios on the API level before it is merged.

We know about `b2_fuse` and we can synchronize with their developers, but there may be other projects that can start using the API layer at any point without discussing it with us, so we must be careful for that, or there is a risk... Which in this particular case can quite easily lead to data loss.